packages/core/src/utils.js

```diff
@@ -671,11 +671,12 @@ export function retrieveSchema(schema, rootSchema = {}, formData = {}) {
     resolvedSchema.hasOwnProperty("additionalProperties") &&
     resolvedSchema.additionalProperties !== false;
   if (hasAdditionalProperties) {
-    return stubExistingAdditionalProperties(
+    stubExistingAdditionalProperties(
```

Comment:
Try doing something like this:
```suggestion
    schema = stubExistingAdditionalProperties(
```

because the change you made just makes `stubExistingAdditionalProperties` not do anything at all, thus breaking the additional properties functionality.